client.go

```diff
@@ -267,8 +267,11 @@ func (c *Client) pushBatch(batch *Batch) (httpext.Response, error) {
 		return *httpext.NewResponse(), errors.Wrap(err, "push request failed")
 	}
 	res.Request.Body = ""
+	if IsSuccessfulResponse(res.Status) {
+		err = c.reportMetricsFromBatch(batch)
```

Comment:
It looks like `c.reportMetricsFromBatch(batch)` will never return an error. Do you think it makes sense to remove the return type from the function signature?

Reply:
I agree @chaudum. It's been removed